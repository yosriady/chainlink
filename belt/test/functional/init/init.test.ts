import { join } from 'path'
import { cat, rm, mkdir } from 'shelljs'
import Init from '../../../src/commands/init'

const TEST_PATH = 'test/functional/init/'
const TEST_FS_PATH = join(TEST_PATH, 'testfs')
const RUNTIME_CONFIG_PATH = `${TEST_FS_PATH}/.beltrc`

describe('init tests', () => {
  beforeAll(() => {
    mkdir(TEST_FS_PATH)
  })

  afterAll(() => {
    rm('-r', TEST_FS_PATH)
  })

  it('should write to runtime config', async () => {
    await Init.run([
      "-m 'raise clutch area heavy horn course filter farm deny solid finger sudden' -c 4 -p fdf38d85d15e434e9b2ca152b7b1bc6f",
      TEST_FS_PATH,
    ])

    expect(cat(RUNTIME_CONFIG_PATH).stdout).toMatchSnapshot()
  })

  it('should support partial updates to runtime config', async () => {
    await Init.run(['-p gdf38d85d15e434e9b2ca152b7b1bc6f', TEST_FS_PATH])

    expect(cat(RUNTIME_CONFIG_PATH).stdout).toMatchSnapshot()
  })
})
