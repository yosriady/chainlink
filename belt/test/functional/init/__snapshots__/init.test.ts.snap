// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init tests should support partial updates to runtime config 1`] = `
"{
    \\"chainId\\": 4,
    \\"mnemonic\\": \\" 'raise clutch area heavy horn course filter farm deny solid finger sudden' -c 4 -p fdf38d85d15e434e9b2ca152b7b1bc6f\\",
    \\"infuraProjectId\\": \\" gdf38d85d15e434e9b2ca152b7b1bc6f\\",
    \\"gasPrice\\": 40000000000,
    \\"gasLimit\\": 8000000
}"
`;

exports[`init tests should write to runtime config 1`] = `
"{
    \\"chainId\\": 4,
    \\"mnemonic\\": \\" 'raise clutch area heavy horn course filter farm deny solid finger sudden' -c 4 -p fdf38d85d15e434e9b2ca152b7b1bc6f\\",
    \\"infuraProjectId\\": \\"\\",
    \\"gasPrice\\": 40000000000,
    \\"gasLimit\\": 8000000
}"
`;
